include/lmms_math.h

```diff
@@ -360,6 +360,13 @@ static inline int numDigitsAsInt(float f)
 	return digits;
 }
 
+
+static inline bool floatCompare(float x, float y)
```

Comment:
@tresf said on discord that compare functions return an integer value and not a bool according to some convention. Is this something to note? If so, what should I rename to.

Reply:
I agree with him. "Compare" is usually used for functions that determine the order of their arguments, which is often encoded in an `int` (where a negative value indicates the first is less than the second, a positive value that it is greater, and zero indicates equality). This function instead tests for approximate equality.

A second point I would add is that in C++ it is often considered a code smell to include types in function names. This is necessary in C as there is no function overloading, but serves little purpose in C++ and can be detrimental in some cases. See [the Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#nl5-avoid-encoding-type-information-in-names).

How about `isClose`, or if you really want to stress that this is for floating-point values (but not specifically `float`s), `fpClose`?

Reply:
I would avoid "equal" as it doesn't check for equality, and doesn't even define an equivalence relation. "Check" is redundant: what else is it going to do - guess? Also, "check" could be interpreted as meaning "require" or "make sure", suggesting that the function wraps an assertion, or is a macro for an early return (I've seen it used in both of these ways before).

The standard library tends to be quite terse and to name functions only after the property they determine, e.g. `container.empty()` instead of `container.check_empty()`, `std::any` instead of `std::check_for_any`, `std::equal` instead of `std::test_equal`. I was tempted to suggest simply using `close` for the name, but that word has several meanings in English, and could be confused for something you might do to a file or a window. That's my reason for adding "is" (to indicate that "close" is an adjective, not a verb), or "fp" (for additional context).

Reply:
For some reason, `isClose` seems ambiguous. I'll use the old name, `isEqual` instead.

Reply:
I won't object to using the old name, since it doesn't make anything worse, but I still think it's an inaccurate description of the function. What seems ambiguous about `isClose`?

Reply:
Well yeah, it is close. I'll do the change. I was just confused.

Reply:
I thought the function was checking for equality rather than closeness

Reply:
> I thought the function was checking for equality rather than closeness

`==` is equality.  `std::abs(x - y) < F_EPSILON` is "close enough".

Reply:
Yup that was the source for my confusion. It's clear now.

Reply:
A fantastic write-up on this problem: https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/

... of which I found by reading the comments of [this S.O. answer](https://stackoverflow.com/a/17341/3196753).  I really wasn't looking for a technical solution, but rather a good name for this function.

I asked A.I. and it reiterated what's already been said, quoting:

> #### Short and Descriptive:
> * `areAlmostEqual`
> * `isCloseTo`
> * `approximatelyEqual`
> 
> #### More Specific (if you need to indicate tolerance):
> * `equalsWithinTolerance`
> * `withinEpsilon`
>
> #### Alternative (if you prefer a verb-based name):
> * `compareFloats` (but this might be less clear about the return value)

"approximatelyEqual" is the most descriptive in my opinion.  "withinEpsilon" sounds the coolest. 😎

Reply:
`approximatelyEqual` seems ideal to me personally, since I feel like the idea for this function is to check if `x` and `y` are practically equal. `isClose` and `isCloseTo` might be a problem since there isn't anything to say how close. Equal, on the other hand, says "close enough to be considered equal" which is more descriptive in my opinion.

Reply:
I changed the name to `approximatelyEqual`